analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,296 @@
+stopifnot(identical(
+  sum(is.na(select(htl_df, Kids_First_Biospecimen_ID,
+                   Kids_First_Participant_ID))),
+  as.integer(0)))
+
+fusion_df <- read_tsv(file.path(root_dir,
+  'analyses/fusion_filtering/results/pbta-fusion-putative-oncogenic.tsv'))
+# assert all records have Sample
+stopifnot(identical(sum(is.na(fusion_df$Sample)), as.integer(0)))
+
+# primary independent sample data frame
+primary_indp_sdf <- read_tsv(file.path(
+  root_dir,
+  'analyses/independent-samples/results/independent-specimens.wgs.primary.tsv'),
+  col_types = cols(
+    .default = col_guess()))
+
+# relapse independent samples
+relapse_indp_sdf <- read_tsv(file.path(
+  root_dir,
+  'analyses/independent-samples/results/independent-specimens.wgs.relapse.tsv'),
+  col_types = cols(
+    .default = col_guess()))
+
+#efo cancer_group mappings
+efo_mondo_cg_df <- read_tsv(file.path(data_dir,'efo-mondo-map.tsv'),
+                            col_types = cols(.default = col_guess())) %>%
+  distinct()
+# efo_mondo_cg_df$cancer_group[
+#   !efo_mondo_cg_df$cancer_group %in% htl_df$cancer_group]
+
+# assert all cancer_groups are not NA
+stopifnot(identical(sum(is.na(efo_mondo_cg_df$cancer_group)), as.integer(0)))
+# assert all cancer_groups are unique.
+# result SNV table is left joined by cancer_groups
+stopifnot(identical(length(unique(efo_mondo_cg_df$cancer_group)),
+                    nrow(efo_mondo_cg_df)))
+
+# ensg hugo rmtl mappings
+ensg_hugo_rmtl_df <- read_tsv(file.path(data_dir,'ensg-hugo-rmtl-v1-mapping.tsv'),
+                              col_types = cols(.default = col_guess())) %>%
+  distinct()
+# assert all ensg_ids and gene_symbols are not NA
+stopifnot(identical(sum(is.na(ensg_hugo_rmtl_df$ensg_id)), as.integer(0)))
+stopifnot(identical(sum(is.na(ensg_hugo_rmtl_df$gene_symbol)), as.integer(0)))
+# assert all ensg_id are unique
+# result SNV table is left joined by ensg_id
+stopifnot(identical(length(unique(ensg_hugo_rmtl_df$ensg_id)),
+                    nrow(ensg_hugo_rmtl_df)))
+
+
+
+# Subset independent samples in histology table --------------------------------
+message('Prepare data...')
+# td = tumor descriptor
+td_htl_dfs <- list(
+  primary_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             primary_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  relapse_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             relapse_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  overall_htl_df = htl_df %>%
+    filter(sample_type == 'Tumor')
+)
+
+# keep only samples with non-NA cancer_group and cohort
+# add cancer_group_cohort column
+td_htl_dfs <- lapply(td_htl_dfs, function(x) {
+  # assert all Kids_First_Biospecimen_IDs are unique
+  stopifnot(identical(nrow(x), length(unique(x$Kids_First_Biospecimen_ID))))
+  
+  fx <- filter(x, !is.na(cancer_group), !is.na(cohort))
+  fx <- mutate(
+    fx, cancer_group_cohort = paste(cancer_group, cohort, sep = '___'))
```

Comment:
```suggestion
```

The `cancer_group_cohort` column is no longer used in the latest commit of <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/45> .

Maybe delete them, and see if the results are still identical?